laceworksdk/api/base_endpoint.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class BaseEndpoint(object):
+    """
+    Lacework BaseEndpoint Class.
+    """
+
+    def __init__(self,
+                 session,
+                 endpoint_path,
+                 object_type):
+        """
+        :param session: An instance of the HttpSession class.
+        :param endpoint_path: The URL endpoint to use for the object type.
+        :param object_type: The Lacework object type to use for validation. (TODO)
+        """
+
+        super().__init__()
+        self._session = session
+        self._endpoint_path = endpoint_path
+        self._object_type = object_type
+
+    def build_dict_from_items(self, *dicts, **items):
+        """
+        A method to build a dictionary based on inputs, pruning items that are None
+        """
+
+        dict_list = list(dicts)
+        dict_list.append(items)
+        result = {}
+
+        for d in dict_list:
+            for key, value in d.items():
+                camel_key = self._convert_lower_camel_case(key)
+                if value is not None:
+                    result[camel_key] = value
+
+        return result
+
+    def build_url(self, id=None, resource=None):
+        """
+        Builds the URL to use based on the endpoint path, resource, and ID.
+        """
+
+        if resource:
+            if id:
+                endpoint_url = f"{self._endpoint_path}/{resource}/{id}"
+            else:
+                endpoint_url = f"{self._endpoint_path}/{resource}"
+        else:
+            if id:
+                endpoint_url = f"{self._endpoint_path}/{id}"
+            else:
+                endpoint_url = self._endpoint_path
+
+        return endpoint_url
+
+    @staticmethod
+    def _convert_lower_camel_case(param_name):
+        """
+        Convert a Pythonic variable name to lowerCamelCase.
+        """
+
+        return param_name.split('_')[0] + ''.join(x.capitalize() or '_' for x in param_name.split('_')[1:])
```

Comment:
this is a really long one-liner that is hard to read... can you split this up into more lines....

```python
words = param_name.split('_')
first_word = words[0]

if len(words) == 1:
  return first_word

word_string = "".join([x.capitalize() or "_" for x in words[1:]])

return f"{first_word}{word_string}"
```

or something like that... makes it a bit easier to read and then debug if it fails...

(also you use single quotes here instead of double)